src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs

```diff
@@ -169,9 +186,14 @@ internal static object CreateObjectFromJsonElement(JsonElement jsonElement)
                 foreach (JsonProperty property in jsonElement.EnumerateObject())
                     numItems++;
 
+                int index = 0;
                 KeyValuePair<string, object>[] kvps = new KeyValuePair<string, object>[numItems];
                 foreach (JsonProperty property in jsonElement.EnumerateObject())
-                    kvps[numItems++] = new KeyValuePair<string, object>(property.Name, CreateObjectFromJsonElement(property.Value));
+                {
+                    currentDepth++;
```

Comment:
Does mean that the Json can only have MaxDepth properties {"string":"value"} ... as the check is when the method is entered.
Should we differentiate between Object and Array?

Reply:
what do you mean by:

Should we differentiate between Object and Array?

Reply:
I think the PR is fine, the currentDepth-- was under the comments and i missed it.
I agree with Westin that just passing currnetDepth+1 would be clearer.